[mypy] Fix Optional type check

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62668
* #62667

Differential Revision: [D30077960](https://our.internmc.facebook.com/intern/diff/D30077960)

IvanKobzarev committed:
[mypy] Fix Optional type check

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30077960).

842974287 commented on torch/fx/experimental/fx2trt/fx2trt.py:
```diff
@@ -287,6 +287,7 @@ def run_node(self, n):
         return super().run_node(n)
 
     def placeholder(self, target, args, kwargs):
+        assert self.optimization_profiles
```
We probably want to add it below line 299 because `self.optimization_profiles` could be None in some of the cases.

842974287 reviewed (commented):
Thanks for the fix! So sorry for the inconvenience.

IvanKobzarev committed:
Update on "[mypy] Fix Optional type check"


Differential Revision: [D30077960](https://our.internmc.facebook.com/intern/diff/D30077960)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30077960).
